src/modals/send-modal/send-modal.tsx

```diff
@@ -80,87 +61,39 @@ export const SendModal: FC = () => {
     receiverPublicKeyHash,
     ownAccount,
     transferBetweenOwnAccounts,
-    amount
-  }: SendModalFormValues) =>
-    void (
-      isDefined(amount) &&
+    amount,
+    amountUnitIndex
+  }: SendModalFormValues) => {
+    const exchangeRateKey = token.symbol === TEZ_TOKEN_METADATA.symbol ? TEZ_TOKEN_METADATA.name : getTokenSlug(token);
```

Comment:
there are a lot of the same logic, let's create util that will return `exchangeRateKey` by token and reuse it everywhere